File: src/main/java/dev/akif/cookbook/recipe/Recipe.java
package dev.akif.cookbook.recipe;

import dev.akif.cookbook.ingredient.MeasuredIngredient;

import java.util.List;
import java.util.Map;

public record Recipe(
        long id,
        String name,
        int numberOfServings,
        Map<String, List<MeasuredIngredient>> ingredients,
        List<String> instructions
) {
}

File: src/main/kotlin/dev/akif/cookbook/recipe/Recipe.kt
package dev.akif.cookbook.recipe

import dev.akif.cookbook.ingredient.MeasuredIngredient

data class Recipe(
    val id: Long,
    val name: String,
    val numberOfServings: Int,
    val ingredients: Map<String, List<MeasuredIngredient>>,
    val instructions: List<String>
)
